app/components/alchemy/admin/resource/table.rb

```diff
@@ -0,0 +1,155 @@
+                        <%= view_context.capture(row, &column.block) %>
+                      </td>
+                    <% end %>
+                    <% if actions.present? %>
+                      <td class="tools">
+                        <% actions.each do |action| %>
+                          <% if action.name.nil? || can?(action.name, row) %>
+                            <% if action.tooltip.present? %>
+                              <sl-tooltip content="<%= action.tooltip %>">
+                                <%= view_context.capture(row, &action.block) %>
+                              </sl-tooltip>
+                            <% else %>
+                              <%= view_context.capture(row, &action.block) %>
+                            <% end %>
+                          <% end %>
+                        <% end %>
+                      </td>
+                    <% end %>
+                  </tr>
+                <% end %>
+              </tbody>
+            </table>
+          <% else %>
+            <div class="info">
+              <%= render_icon('info') %>
+              <%= nothing_found_label %>
+            </div>
+          <% end %>
+        ERB
+
+        def initialize(collection, ransack_query: nil, nothing_found_label: Alchemy.t("Nothing found"), resource_url: :resource_url_proxy)
+          @collection = collection
+          @ransack_query = ransack_query
+          @nothing_found_label = nothing_found_label
+          @resource_url = resource_url
+          @columns = []
+          @actions = []
+        end
+
+        def column(name, label: nil, sortable: false, type: :string, &block)
+          @columns << Column.new(name, label: label, sortable: sortable, type: type, &block)
+        end
+
+        def icon_column(icon = nil)
+          @columns << Column.new(:icon, label: "") do |row|
+            render_icon(icon || yield(row), size: "xl")
+          end
+        end
+
+        def action(name = nil, tooltip: nil, &block)
+          @actions << Action.new(name, tooltip: tooltip, &block)
+        end
+
+        def delete_action(tooltip: Alchemy.t("Delete"), search_filter_params: {})
+          action(:destroy, tooltip: tooltip) do |row|
+            delete_button resource_path(row, search_filter_params)
+          end
+        end
+
+        def edit_action(tooltip: Alchemy.t("Edit"), search_filter_params: {})
+          action(:edit, tooltip: tooltip) do |row|
+            link_to_dialog render_icon(:edit),
+              edit_resource_path(row, search_filter_params),
+              {
+                title: Alchemy.t("Edit"),
+                size: resource_window_size
+              },
+              class: "icon_button"
+          end
+        end
+
+        private
+
+        ##
+        # the before_render - method is necessary to force ViewComponent to evaluate the add_column - calls
+        def before_render
+          content
+        end
+
+        class Column
```

Comment:
Could this be a `ResourceAttribute` (`ResourceDisplay`, etc..) so we could use this in future resource views that are not tables?

Or put this in a parent object that holds common things and implement a table column that inherits

Reply:
At the moment it is called attribute.